File: Anul 2/Medii de Programare si Proiectare/mpp-proiect-java-Mothyque/backend/Server/src/main/java/org/example/service/UserService.java
package org.example.service;

import org.example.domain.User;
import org.example.repository.Repository;
import org.example.repository.UserRepository;

import java.security.MessageDigest;

public class UserService extends Service<Integer, User>
{
    public UserService(Repository<Integer, User> repository)
    {
        super(repository);
    }

    public User findByUsername(String username)
    {
        return ((UserRepository) repository).findByUsername(username);
    }

    public User authenticate(String username, String password)
    {
        User user = findByUsername(username);
        if (user != null && user.getPassword().equals(hashPassword(password)))
        {
            return user;
        }
        return null;
    }

    public String hashPassword(String password)
    {
        try
        {
            MessageDigest md = MessageDigest.getInstance("SHA-1");
            md.update(password.getBytes());
            byte[] digest = md.digest();
            StringBuilder sb = new StringBuilder();
            for (byte b : digest)
            {
                sb.append(String.format("%02x", b));
            }
            return sb.toString();
        }
        catch (Exception e)
        {
            throw new RuntimeException("Error hashing password", e);
        }
    }
}

File: Anul 2/Medii de Programare si Proiectare/Lab/mpp-proiect-csharp-Mothyque/ProiectMPP/ProiectMPP.Server/Service/UserService.cs
using ProiectMPP.Common.Domain;
using ProiectMPP.Server.Repository;
using System;
using System.Security.Cryptography;
using System.Text;

namespace ProiectMPP.Server.Service
{
    public class UserService : Service<int, User>
    {
        public UserService(IUserRepository repository) : base(repository)
        {
        }

        public User FindByUsername(string username)
        {
            return ((IUserRepository)repository).FindByUsername(username);
        }

        public User Authenticate(string username, string password)
        {
            User user = FindByUsername(username);
            if (user != null && user.Password.Equals(HashPassword(password)))
            {
                return user;
            }
            return null;
        }

        public string HashPassword(string password)
        {
            if (string.IsNullOrEmpty(password)) return string.Empty;

            using (SHA1 sha1 = SHA1.Create())
            {
                byte[] inputBytes = Encoding.UTF8.GetBytes(password);
                byte[] hashBytes = sha1.ComputeHash(inputBytes);
                StringBuilder sb = new StringBuilder();
                foreach (byte b in hashBytes)
                {
                    sb.Append(b.ToString("x2"));
                }
                return sb.ToString();
            }
        }
    }
}